src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -736,7 +736,7 @@ def acquireTiledArea(streams, stage, area, overlap=0.2, settings_obs=None, log_p
     if not mem_sufficient:
         raise IOError("Not enough RAM to safely acquire the overview: %g GB needed" % (mem_est / 1024 ** 3,))
 
-    future.set_progress(end=mem_est + time.time())
+    future.set_progress(end=task.estimateTime() + time.time())
```

Comment:
Do futures have a timeout? Or maybe some function which you can connect to a timeout? For example if updating the live image takes longer than the actual scan you want the live update future to go into timeout at the moment the scan is finished. (I think for the live update itself this wouldn't be useful but it is the first example which came to my mind)
Not related to this PR but maybe we want such a feature integrated in the future in the future(class).

Reply:
Yes, there is such a thing. You can call `future.result(timeout)`, in which case it'll raise an error if the tasks takes more than you expect. With CancellableFutures (special on Odemis), you then can decide to stop the task. (Otherwise, the only thing you can really do is to leave the task running in the background while you're not waiting for it).